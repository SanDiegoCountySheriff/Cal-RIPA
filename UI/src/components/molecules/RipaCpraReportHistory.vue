<template>
  <v-container class="tw-mt-2">
    <v-data-table :loading="loading" :items="reports" :headers="headers">
      <template v-slot:[`item.actions`]="{ item }">
        <v-btn color="primary" small @click="downloadCpraReport(item.fileName)"
          >Download</v-btn
        >
      </template>
    </v-data-table>
  </v-container>
</template>

<script>
export default {
  name: 'ripa-cpra-report-history',

  data() {
    return {
      reports: [],
      headers: [
        { text: 'From Date', value: 'fromDate' },
        { text: 'To Date', value: 'toDate' },
        { text: 'Officer Name', value: 'officerName' },
        { text: 'Download', value: 'actions', sortable: false },
      ],
    }
  },

  methods: {
    init() {
      this.reports = this.items
    },

    downloadCpraReport(fileName) {
      this.$emit('handleDownloadCpraReport', fileName)
    },
  },

  created() {
    this.init()
  },

  watch: {
    items(val) {
      this.reports = val
    },
  },

  props: {
    loading: {
      type: Boolean,
      default: false,
    },

    items: {
      type: Array,
      default: () => [],
    },
  },
}
</script>
