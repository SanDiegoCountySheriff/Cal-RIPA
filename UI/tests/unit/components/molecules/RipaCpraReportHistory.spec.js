import RipaCpraReportHistory from '@/components/molecules/RipaCpraReportHistory.vue'
import { shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'

describe('Ripa CPRA Report History', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaCpraReportHistory, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = factory()

    expect(wrapper.element).toMatchSnapshot()
  })
})
