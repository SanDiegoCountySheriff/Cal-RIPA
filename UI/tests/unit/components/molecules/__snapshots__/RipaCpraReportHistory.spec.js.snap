// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa CPRA Report History should match snapshot 1`] = `
<v-container-stub
  class="tw-mt-2"
  tag="div"
>
  <v-data-table-stub
    customfilter="[Function]"
    customgroup="[Function]"
    customsort="[Function]"
    expanded=""
    expandicon="$expand"
    groupby=""
    groupdesc=""
    headers="[object Object],[object Object],[object Object],[object Object]"
    itemclass=""
    itemkey="id"
    items=""
    itemsperpage="10"
    loaderheight="4"
    loadingtext="$vuetify.dataIterator.loadingText"
    locale="en-US"
    mobilebreakpoint="600"
    nodatatext="$vuetify.noDataText"
    noresultstext="$vuetify.dataIterator.noResultsText"
    options="[object Object]"
    page="1"
    selectablekey="isSelectable"
    serveritemslength="-1"
    sortby=""
    sortdesc=""
    value=""
  />
</v-container-stub>
`;
